### ISSUE TITLE
Documentation publishing and guidelines

### ISSUE THREAD

**Issue by mbrzakovic** [rapideditor/rapid-sdk#41]
Currently math package has code documentation written in // and osm module has /***/.

I suggest we standardize how we document our code,

I am new at this but here are some of the resources I've found: 
https://tsdoc.org
https://typedoc.org
https://jsdoc.app/index.html

Stretch:
Ideally, we should write code documentation only once, and then use some generators to craft read me sections automatically.

**Comment by bhousel** [comment]
I agree it would be great to document once and have it generated elsewhere in the READMEs or wherever else docs will live.

I originally did use jsdoc comments for this project but I removed it all in 32c156d , mostly because at the time I felt that I was fighting with the tools / spending too much time tweaking themes and layout.  But I have changed my thinking on this in the past few years, and I agree we should put them back.
